### PR TITLE
Update patch to fix a vaapi buffer sync regression

### LINKED ISSUE #99: Playback broken with jellyfin-androidtv
I've installed jellyfin-stable on my nas (Ubuntu 20.04 with HWE-stack) from your repository. It's installed directly, no container or vm. Since the last jellyfin-ffmpeg update on 28.02, jellyfin-androidtv (0.12.3) wants to transcode many videos (unnecessarily) and playback fails (with ExoPlayer forced, libVLC plays, but is unusable, because of missing passthrough). Before the update, transcoding was not necessary and direct playback worked. The jellyfin-ffmpeg update was the only jellyfin update in the last time, so this update should be the reason.

**a few apt-cache results**
```
jellyfin:
  Installed: 10.7.7-1
  Candidate: 10.7.7-1
  Version table:
 *** 10.7.7-1 500
        500 https://repo.jellyfin.org/ubuntu focal/main amd64 Packages
        100 /var/lib/dpkg/status

jellyfin-ffmpeg:
  Installed: 4.4.1-2-focal
  Candidate: 4.4.1-2-focal
  Version table:
 *** 4.4.1-2-focal 500

libva2:
  Installed: 2.7.0-2
  Candidate: 2.7.0-2
  Version table:
 *** 2.7.0-2 500
        500 http://de.archive.ubuntu.com/ubuntu focal/universe amd64 Packages
        100 /var/lib/dpkg/status

intel-media-va-driver-non-free:
  Installed: 20.1.1+ds1-1build1
  Candidate: 20.1.1+ds1-1build1
  Version table:
 *** 20.1.1+ds1-1build1 500
        500 http://de.archive.ubuntu.com/ubuntu focal/multiverse amd64 Packages
        100 /var/lib/dpkg/status
```

**uname -a**
```
Linux nas 5.13.0-30-generic #33~20.04.1-Ubuntu SMP Mon Feb 7 14:25:10 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

**vainfo**
```
libva info: VA-API version 1.7.0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/iHD_drv_video.so
libva info: Found init function __vaDriverInit_1_7
libva error: /usr/lib/x86_64-linux-gnu/dri/iHD_drv_video.so init failed
libva info: va_openDriver() returns 1
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/i965_drv_video.so
libva info: Found init function __vaDriverInit_1_6
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.7 (libva 2.6.0)
vainfo: Driver version: Intel i965 driver for Intel(R) CherryView - 2.4.0
vainfo: Supported profile and entrypoints
      VAProfileMPEG2Simple            : VAEntrypointVLD
      VAProfileMPEG2Simple            : VAEntrypointEncSlice
      VAProfileMPEG2Main              : VAEntrypointVLD
      VAProfileMPEG2Main              : VAEntrypointEncSlice
      VAProfileH264ConstrainedBaseline: VAEntrypointVLD
      VAProfileH264ConstrainedBaseline: VAEntrypointEncSlice
      VAProfileH264Main               : VAEntrypointVLD
      VAProfileH264Main               : VAEntrypointEncSlice
      VAProfileH264High               : VAEntrypointVLD
      VAProfileH264High               : VAEntrypointEncSlice
      VAProfileH264MultiviewHigh      : VAEntrypointVLD
      VAProfileH264MultiviewHigh      : VAEntrypointEncSlice
      VAProfileH264StereoHigh         : VAEntrypointVLD
      VAProfileH264StereoHigh         : VAEntrypointEncSlice
      VAProfileVC1Simple              : VAEntrypointVLD
      VAProfileVC1Main                : VAEntrypointVLD
      VAProfileVC1Advanced            : VAEntrypointVLD
      VAProfileNone                   : VAEntrypointVideoProc
      VAProfileJPEGBaseline           : VAEntrypointVLD
      VAProfileJPEGBaseline           : VAEntrypointEncPicture
      VAProfileVP8Version0_3          : VAEntrypointVLD
      VAProfileVP8Version0_3          : VAEntrypointEncSlice
      VAProfileHEVCMain               : VAEntrypointVLD
```

**transcoding logs:**
**1)**
```
/videos/cec95a09-be1c-d980-0eae-bb82e144b6a6/stream.mkv

{"Protocol":0,"Id":"cec95a09be1cd9800eaebb82e144b6a6","Path":"/srv/shares/public/video/Filme/Beast.mkv","EncoderPath":null,"EncoderProtocol":null,"Type":0,"Container":"mkv,webm","Size":8156379310,"Name":"Beast","IsRemote":false,"ETag":"1f16b8835f0b9af7de26853937452af8","RunTimeTicks":64184000000,"ReadAtNativeFramerate":false,"IgnoreDts":false,"IgnoreIndex":false,"GenPtsInput":false,"SupportsTranscoding":true,"SupportsDirectStream":true,"SupportsDirectPlay":true,"IsInfiniteStream":false,"RequiresOpening":false,"OpenToken":null,"RequiresClosing":false,"LiveStreamId":null,"BufferMs":null,"RequiresLooping":false,"SupportsProbing":true,"VideoType":0,"IsoType":null,"Video3DFormat":null,"MediaStreams":[{"Codec":"h264","CodecTag":null,"Language":null,"ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"Comment":null,"TimeBase":"1/1000","CodecTimeBase":null,"Title":null,"VideoRange":"SDR","localizedUndefined":null,"localizedDefault":null,"localizedForced":null,"DisplayTitle":"1080p H264 SDR","NalLengthSize":"0","IsInterlaced":false,"IsAVC":false,"ChannelLayout":null,"BitRate":10166246,"BitDepth":8,"RefFrames":1,"PacketLength":null,"Channels":null,"SampleRate":null,"IsDefault":true,"IsForced":false,"Height":804,"Width":1920,"AverageFrameRate":24,"RealFrameRate":24,"Profile":"High","Type":1,"AspectRatio":"2.40:1","Index":0,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":"yuv420p","Level":41,"IsAnamorphic":null},{"Codec":"dts","CodecTag":null,"Language":"ger","ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"Comment":null,"TimeBase":"1/1000","CodecTimeBase":null,"Title":null,"VideoRange":null,"localizedUndefined":null,"localizedDefault":null,"localizedForced":null,"DisplayTitle":"Ger - DTS - 5.1 - Default","NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":"5.1","BitRate":1536000,"BitDepth":null,"RefFrames":null,"PacketLength":null,"Channels":6,"SampleRate":48000,"IsDefault":true,"IsForced":false,"Height":null,"Width":null,"AverageFrameRate":null,"RealFrameRate":null,"Profile":"DTS","Type":0,"AspectRatio":null,"Index":1,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":null,"Level":0,"IsAnamorphic":null},{"Codec":"dts","CodecTag":null,"Language":"eng","ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"Comment":null,"TimeBase":"1/1000","CodecTimeBase":null,"Title":null,"VideoRange":null,"localizedUndefined":null,"localizedDefault":null,"localizedForced":null,"DisplayTitle":"En - DTS - 5.1","NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":"5.1","BitRate":1536000,"BitDepth":null,"RefFrames":null,"PacketLength":null,"Channels":6,"SampleRate":48000,"IsDefault":false,"IsForced":false,"Height":null,"Width":null,"AverageFrameRate":null,"RealFrameRate":null,"Profile":"DTS","Type":0,"AspectRatio":null,"Index":2,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":null,"Level":0,"IsAnamorphic":null},{"Codec":"PGSSUB","CodecTag":null,"Language":"ger","ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"Comment":null,"TimeBase":"1/1000","CodecTimeBase":null,"Title":"forced","VideoRange":null,"localizedUndefined":"Undefiniert","localizedDefault":"Standard","localizedForced":"Erzwungen","DisplayTitle":"forced - Ger - Erzwungen","NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":null,"BitRate":null,"BitDepth":null,"RefFrames":null,"PacketLength":null,"Channels":null,"SampleRate":null,"IsDefault":false,"IsForced":true,"Height":null,"Width":null,"AverageFrameRate":null,"RealFrameRate":null,"Profile":null,"Type":2,"AspectRatio":null,"Index":3,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":null,"Level":0,"IsAnamorphic":null},{"Codec":"PGSSUB","CodecTag":null,"Language":"ger","ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"Comment":null,"TimeBase":"1/1000","CodecTimeBase":null,"Title":"complete","VideoRange":null,"localizedUndefined":"Undefiniert","localizedDefault":"Standard","localizedForced":"Erzwungen","DisplayTitle":"complete - Ger","NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":null,"BitRate":null,"BitDepth":null,"RefFrames":null,"PacketLength":null,"Channels":null,"SampleRate":null,"IsDefault":false,"IsForced":false,"Height":null,"Width":null,"AverageFrameRate":null,"RealFrameRate":null,"Profile":null,"Type":2,"AspectRatio":null,"Index":4,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":null,"Level":0,"IsAnamorphic":null}],"MediaAttachments":[],"Formats":[],"Bitrate":13238246,"Timestamp":null,"RequiredHttpHeaders":{},"TranscodingUrl":null,"TranscodingSubProtocol":null,"TranscodingContainer":null,"AnalyzeDurationMs":null,"DefaultAudioStreamIndex":null,"DefaultSubtitleStreamIndex":null}

/usr/lib/jellyfin-ffmpeg/ffmpeg -hwaccel vaapi -hwaccel_output_format vaapi -vaapi_device /dev/dri/renderD128 -i file:"/srv/shares/public/video/Filme/Beast.mkv" -map 0:0 -map 0:1 -codec:v:0 h264_vaapi -force_key_frames "expr:gte(t,n_forced*5)" -filter_complex "[0:3]scale=1920x804[sub];[0:0]format=nv12|vaapi,hwupload,scale_vaapi=format=nv12,hwdownload[base];[base][sub]overlay,format=nv12,hwupload" -copyts -avoid_negative_ts disabled -start_at_zero -b:v 10166246 -maxrate 10166246 -bufsize 20332492 -profile:v:0 high -level 41 -vsync -1 -map_metadata -1 -map_chapters -1 -threads 0 -codec:a:0 ac3 -ac 6 -ab 640000 -y "/var/lib/jellyfin/transcodes/0a2902b36036106b30c13afb47fc689d.mkv"


ffmpeg version 4.4.1-Jellyfin Copyright (c) 2000-2021 the FFmpeg developers
  built with gcc 9 (Ubuntu 9.3.0-17ubuntu1~20.04)
  configuration: --prefix=/usr/lib/jellyfin-ffmpeg --target-os=linux --extra-version=Jellyfin --disable-doc --disable-ffplay --disable-shared --disable-libxcb --disable-sdl2 --disable-xlib --enable-lto --enable-gpl --enable-version3 --enable-static --enable-gmp --enable-gnutls --enable-libdrm --enable-libass --enable-libfreetype --enable-libfribidi --enable-libfontconfig --enable-libbluray --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libdav1d --enable-libwebp --enable-libvpx --enable-libx264 --enable-libx265 --enable-libzvbi --enable-libzimg --arch=amd64 --enable-opencl --enable-vaapi --enable-amf --enable-libmfx --enable-vdpau --enable-ffnvcodec --enable-cuda --enable-cuda-llvm --enable-cuvid --enable-nvdec --enable-nvenc
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100
[AVHWDeviceContext @ 0x5601fb8ca640] libva: /usr/lib/jellyfin-ffmpeg/lib/dri/iHD_drv_video.so init failed
[matroska,webm @ 0x5601fb8e7a00] Could not find codec parameters for stream 3 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x5601fb8e7a00] Could not find codec parameters for stream 4 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
Input #0, matroska,webm, from 'file:/srv/shares/public/video/Filme/Beast.mkv':
  Metadata:
    encoder         : libebml v1.3.6 + libmatroska v1.4.9
    creation_time   : 2019-11-05T16:28:20.000000Z
  Duration: 01:46:58.40, start: 0.000000, bitrate: 10166 kb/s
  Chapters:
    Chapter #0:0: start 0.000000, end 536.500000
      Metadata:
        title           : Chapter 01
    Chapter #0:1: start 536.500000, end 1104.541000
      Metadata:
        title           : Chapter 02
    Chapter #0:2: start 1104.541000, end 1788.083000
      Metadata:
        title           : Chapter 03
    Chapter #0:3: start 1788.083000, end 2298.125000
      Metadata:
        title           : Chapter 04
    Chapter #0:4: start 2298.125000, end 3005.750000
      Metadata:
        title           : Chapter 05
    Chapter #0:5: start 3005.750000, end 3568.500000
      Metadata:
        title           : Chapter 06
    Chapter #0:6: start 3568.500000, end 4042.291000
      Metadata:
        title           : Chapter 07
    Chapter #0:7: start 4042.291000, end 4595.750000
      Metadata:
        title           : Chapter 08
    Chapter #0:8: start 4595.750000, end 5111.291000
      Metadata:
        title           : Chapter 09
    Chapter #0:9: start 5111.291000, end 5623.958000
      Metadata:
        title           : Chapter 10
    Chapter #0:10: start 5623.958000, end 6196.166000
      Metadata:
        title           : Chapter 11
    Chapter #0:11: start 6196.166000, end 6418.400000
      Metadata:
        title           : Chapter 12
  Stream #0:0: Video: h264 (High), yuv420p(progressive), 1920x804 [SAR 1:1 DAR 160:67], 24 fps, 24 tbr, 1k tbn, 48 tbc (default)
    Metadata:
      BPS-eng         : 7135790
      DURATION-eng    : 01:46:58.375000000
      NUMBER_OF_FRAMES-eng: 154041
      NUMBER_OF_BYTES-eng: 5725022582
      _STATISTICS_WRITING_APP-eng: mkvmerge v29.0.0 ('Like It Or Not') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2019-11-05 16:28:20
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
  Stream #0:1(ger): Audio: dts (DTS), 48000 Hz, 5.1(side), fltp, 1536 kb/s (default)
    Metadata:
      BPS-eng         : 1509000
      DURATION-eng    : 01:46:58.400000000
      NUMBER_OF_FRAMES-eng: 601725
      NUMBER_OF_BYTES-eng: 1210670700
      _STATISTICS_WRITING_APP-eng: mkvmerge v29.0.0 ('Like It Or Not') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2019-11-05 16:28:20
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
  Stream #0:2(eng): Audio: dts (DTS), 48000 Hz, 5.1(side), fltp, 1536 kb/s
    Metadata:
      BPS-eng         : 1509000
      DURATION-eng    : 01:46:58.400000000
      NUMBER_OF_FRAMES-eng: 601725
      NUMBER_OF_BYTES-eng: 1210670700
      _STATISTICS_WRITING_APP-eng: mkvmerge v29.0.0 ('Like It Or Not') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2019-11-05 16:28:20
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
  Stream #0:3(ger): Subtitle: hdmv_pgs_subtitle (forced)
    Metadata:
      title           : forced
      BPS-eng         : 238
      DURATION-eng    : 00:25:55.833000000
      NUMBER_OF_FRAMES-eng: 4
      NUMBER_OF_BYTES-eng: 46419
      _STATISTICS_WRITING_APP-eng: mkvmerge v29.0.0 ('Like It Or Not') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2019-11-05 16:28:20
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
  Stream #0:4(ger): Subtitle: hdmv_pgs_subtitle
    Metadata:
      title           : complete
      BPS-eng         : 16127
      DURATION-eng    : 01:39:02.750000000
      NUMBER_OF_FRAMES-eng: 1388
      NUMBER_OF_BYTES-eng: 11979887
      _STATISTICS_WRITING_APP-eng: mkvmerge v29.0.0 ('Like It Or Not') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2019-11-05 16:28:20
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
Stream mapping:
  Stream #0:0 (h264) -> format (graph 0)
  Stream #0:3 (pgssub) -> scale (graph 0)
  hwupload (graph 0) -> Stream #0:0 (h264_vaapi)
  Stream #0:1 -> #0:1 (dts (dca) -> ac3 (native))
Press [q] to stop, [?] for help
[matroska,webm @ 0x5601fb8e7a00] sub2video: using 1920x804 canvas
Output #0, matroska, to '/var/lib/jellyfin/transcodes/0a2902b36036106b30c13afb47fc689d.mkv':
  Metadata:
    encoder         : Lavf58.76.100
  Stream #0:0: Video: h264 (High) (H264 / 0x34363248), vaapi_vld(progressive), 1920x804 [SAR 1:1 DAR 160:67], q=2-31, 10166 kb/s, 24 fps, 1k tbn (default)
    Metadata:
      encoder         : Lavc58.134.100 h264_vaapi
  Stream #0:1: Audio: ac3 ([0] [0][0] / 0x2000), 48000 Hz, 5.1, fltp, 640 kb/s (default)
    Metadata:
      encoder         : Lavc58.134.100 ac3
[h264_vaapi @ 0x5601fb918f40] Failed to sync to output buffer completion: 5 (invalid VAContextID).
Video encoding failed
[h264_vaapi @ 0x5601fb918f40] Failed to sync to output buffer completion: 5 (invalid VAContextID).
Conversion failed!
```

**2)**
```
/videos/f4e967ca-0200-6d25-07ba-387ac93d5b00/stream.mkv

{"Protocol":0,"Id":"f4e967ca02006d2507ba387ac93d5b00","Path":"/srv/shares/public/video/Serien/Brockmire/S01/Brockmire-S01E08.mkv","EncoderPath":null,"EncoderProtocol":null,"Type":0,"Container":"mkv,webm","Size":1013444992,"Name":"Brockmire-S01E08","IsRemote":false,"ETag":"c09c104a20444b10471d9b52d5251977","RunTimeTicks":13853840384,"ReadAtNativeFramerate":false,"IgnoreDts":false,"IgnoreIndex":false,"GenPtsInput":false,"SupportsTranscoding":true,"SupportsDirectStream":true,"SupportsDirectPlay":true,"IsInfiniteStream":false,"RequiresOpening":false,"OpenToken":null,"RequiresClosing":false,"LiveStreamId":null,"BufferMs":null,"RequiresLooping":false,"SupportsProbing":true,"VideoType":0,"IsoType":null,"Video3DFormat":null,"MediaStreams":[{"Codec":"mpeg2video","CodecTag":null,"Language":"eng","ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"Comment":null,"TimeBase":"1/1000","CodecTimeBase":null,"Title":null,"VideoRange":"SDR","localizedUndefined":null,"localizedDefault":null,"localizedForced":null,"DisplayTitle":"480p MPEG2VIDEO SDR","NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":null,"BitRate":5852211,"BitDepth":null,"RefFrames":1,"PacketLength":null,"Channels":null,"SampleRate":null,"IsDefault":false,"IsForced":false,"Height":480,"Width":720,"AverageFrameRate":29.97003,"RealFrameRate":29.97003,"Profile":"Main","Type":1,"AspectRatio":"16:9","Index":0,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":"yuv420p","Level":8,"IsAnamorphic":null},{"Codec":"ac3","CodecTag":null,"Language":"eng","ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"Comment":null,"TimeBase":"1/1000","CodecTimeBase":null,"Title":"Surround 5.1","VideoRange":null,"localizedUndefined":null,"localizedDefault":null,"localizedForced":null,"DisplayTitle":"Surround 5.1 - En - Dolby Digital - Default","NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":"5.1","BitRate":448000,"BitDepth":null,"RefFrames":null,"PacketLength":null,"Channels":6,"SampleRate":48000,"IsDefault":true,"IsForced":false,"Height":null,"Width":null,"AverageFrameRate":null,"RealFrameRate":null,"Profile":null,"Type":0,"AspectRatio":null,"Index":1,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":null,"Level":0,"IsAnamorphic":null},{"Codec":"DVDSUB","CodecTag":null,"Language":"eng","ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"Comment":null,"TimeBase":"1/1000","CodecTimeBase":null,"Title":null,"VideoRange":null,"localizedUndefined":"Undefiniert","localizedDefault":"Standard","localizedForced":"Erzwungen","DisplayTitle":"En - Standard","NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":null,"BitRate":null,"BitDepth":null,"RefFrames":null,"PacketLength":null,"Channels":null,"SampleRate":null,"IsDefault":true,"IsForced":false,"Height":null,"Width":null,"AverageFrameRate":null,"RealFrameRate":null,"Profile":null,"Type":2,"AspectRatio":null,"Index":2,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":null,"Level":0,"IsAnamorphic":null}],"MediaAttachments":[],"Formats":[],"Bitrate":6300211,"Timestamp":null,"RequiredHttpHeaders":{},"TranscodingUrl":null,"TranscodingSubProtocol":null,"TranscodingContainer":null,"AnalyzeDurationMs":null,"DefaultAudioStreamIndex":null,"DefaultSubtitleStreamIndex":null}

/usr/lib/jellyfin-ffmpeg/ffmpeg -hwaccel vaapi -hwaccel_output_format vaapi -vaapi_device /dev/dri/renderD128 -i file:"/srv/shares/public/video/Serien/Brockmire/S01/Brockmire-S01E08.mkv" -map 0:0 -map 0:1 -codec:v:0 h264_vaapi -force_key_frames "expr:gte(t,n_forced*5)" -filter_complex "[0:2]scale=720x480[sub];[0:0]format=nv12|vaapi,hwupload,scale_vaapi=format=nv12,hwdownload[base];[base][sub]overlay,format=nv12,hwupload" -copyts -avoid_negative_ts disabled -start_at_zero -b:v 5852211 -maxrate 5852211 -bufsize 11704422 -profile:v:0 high -level 41 -vsync -1 -map_metadata -1 -map_chapters -1 -threads 0 -codec:a:0 ac3 -ac 6 -ab 448000 -y "/var/lib/jellyfin/transcodes/d251ebad042b9572bc391fe7a1ea80e6.mkv"


ffmpeg version 4.4.1-Jellyfin Copyright (c) 2000-2021 the FFmpeg developers
  built with gcc 9 (Ubuntu 9.3.0-17ubuntu1~20.04)
  configuration: --prefix=/usr/lib/jellyfin-ffmpeg --target-os=linux --extra-version=Jellyfin --disable-doc --disable-ffplay --disable-shared --disable-libxcb --disable-sdl2 --disable-xlib --enable-lto --enable-gpl --enable-version3 --enable-static --enable-gmp --enable-gnutls --enable-libdrm --enable-libass --enable-libfreetype --enable-libfribidi --enable-libfontconfig --enable-libbluray --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libdav1d --enable-libwebp --enable-libvpx --enable-libx264 --enable-libx265 --enable-libzvbi --enable-libzimg --arch=amd64 --enable-opencl --enable-vaapi --enable-amf --enable-libmfx --enable-vdpau --enable-ffnvcodec --enable-cuda --enable-cuda-llvm --enable-cuvid --enable-nvdec --enable-nvenc
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100
[AVHWDeviceContext @ 0x560757b1f640] libva: /usr/lib/jellyfin-ffmpeg/lib/dri/iHD_drv_video.so init failed
Input #0, matroska,webm, from 'file:/srv/shares/public/video/Serien/Brockmire/S01/Brockmire-S01E08.mkv':
  Metadata:
    encoder         : libmakemkv v1.16.5 (1.3.10/1.5.2) x86_64-linux-gnu
    creation_time   : 2021-12-05T10:39:14.000000Z
  Duration: 00:23:05.38, start: 0.000000, bitrate: 5852 kb/s
  Chapters:
    Chapter #0:0: start 0.000000, end 391.557833
      Metadata:
        title           : Chapter 01
    Chapter #0:1: start 391.557833, end 646.145500
      Metadata:
        title           : Chapter 02
    Chapter #0:2: start 646.145500, end 871.036833
      Metadata:
        title           : Chapter 03
    Chapter #0:3: start 871.036833, end 1385.384000
      Metadata:
        title           : Chapter 04
  Stream #0:0(eng): Video: mpeg2video (Main), yuv420p(tv, progressive), 720x480 [SAR 32:27 DAR 16:9], SAR 186:157 DAR 279:157, 29.97 fps, 29.97 tbr, 1k tbn, 59.94 tbc
    Metadata:
      BPS-eng         : 5390907
      DURATION-eng    : 00:23:05.384000000
      NUMBER_OF_FRAMES-eng: 33216
      NUMBER_OF_BYTES-eng: 933559635
      SOURCE_ID-eng   : 0100E0
      _STATISTICS_WRITING_APP-eng: MakeMKV v1.16.5 linux(x64-release)
      _STATISTICS_WRITING_DATE_UTC-eng: 2021-12-05 10:39:14
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES SOURCE_ID
    Side data:
      cpb: bitrate max/min/avg: 8500000/0/0 buffer size: 1835008 vbv_delay: N/A
  Stream #0:1(eng): Audio: ac3, 48000 Hz, 5.1(side), fltp, 448 kb/s (default)
    Metadata:
      title           : Surround 5.1
      BPS-eng         : 448000
      DURATION-eng    : 00:23:05.152000000
      NUMBER_OF_FRAMES-eng: 43286
      NUMBER_OF_BYTES-eng: 77568512
      SOURCE_ID-eng   : 0180BD
      _STATISTICS_WRITING_APP-eng: MakeMKV v1.16.5 linux(x64-release)
      _STATISTICS_WRITING_DATE_UTC-eng: 2021-12-05 10:39:14
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES SOURCE_ID
  Stream #0:2(eng): Subtitle: dvd_subtitle, 720x480 (default)
    Metadata:
      BPS-eng         : 8589
      DURATION-eng    : 00:22:51.168111111
      NUMBER_OF_FRAMES-eng: 500
      NUMBER_OF_BYTES-eng: 1472228
      SOURCE_ID-eng   : 0120BD
      _STATISTICS_WRITING_APP-eng: MakeMKV v1.16.5 linux(x64-release)
      _STATISTICS_WRITING_DATE_UTC-eng: 2021-12-05 10:39:14
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES SOURCE_ID
Stream mapping:
  Stream #0:0 (mpeg2video) -> format (graph 0)
  Stream #0:2 (dvdsub) -> scale (graph 0)
  hwupload (graph 0) -> Stream #0:0 (h264_vaapi)
  Stream #0:1 -> #0:1 (ac3 (native) -> ac3 (native))
Press [q] to stop, [?] for help
[matroska,webm @ 0x560757b3ca00] sub2video: using 720x480 canvas
Output #0, matroska, to '/var/lib/jellyfin/transcodes/d251ebad042b9572bc391fe7a1ea80e6.mkv':
  Metadata:
    encoder         : Lavf58.76.100
  Stream #0:0: Video: h264 (High) (H264 / 0x34363248), vaapi_vld(tv, progressive), 720x480 [SAR 186:157 DAR 279:157], q=2-31, 5852 kb/s, 29.97 fps, 1k tbn (default)
    Metadata:
      encoder         : Lavc58.134.100 h264_vaapi
  Stream #0:1: Audio: ac3 ([0] [0][0] / 0x2000), 48000 Hz, 5.1, fltp, 448 kb/s (default)
    Metadata:
      encoder         : Lavc58.134.100 ac3
[h264_vaapi @ 0x560757b88580] Failed to sync to output buffer completion: 5 (invalid VAContextID).
Video encoding failed
[h264_vaapi @ 0x560757b88580] Failed to sync to output buffer completion: 5 (invalid VAContextID).
Conversion failed!
```

### PR DESCRIPTION
**Pull request by nyanmisaka**
Need some testing...

**Issues**
Should fixes #99 #92

### CODE REVIEW

File: debian/patches/0027-add-fixes-to-improve-the-performance-of-vaapi-encode.patch [hunk 1]
```diff
@@ -209,25 +218,29 @@ Index: jellyfin-ffmpeg/libavcodec/vaapi_encode.h
...
      // Slice structure.
      int slice_block_rows;
-@@ -345,6 +348,10 @@ typedef struct VAAPIEncodeContext {
+@@ -345,6 +348,13 @@ typedef struct VAAPIEncodeContext {
      int             roi_warned;
  
      AVFrame         *frame;
 +
++    // Whether the driver support vaSyncBuffer.
++    int             has_sync_buffer_func;
```
Comment (Shadowghost) [inline debian/patches/0027-add-fixes-to-improve-the-performance-of-vaapi-encode.patch:227]:
```suggestion
+    // Whether the driver supports vaSyncBuffer.
+    int has_sync_buffer_func;
```

File: debian/patches/0027-add-fixes-to-improve-the-performance-of-vaapi-encode.patch [hunk 2]
```diff
@@ -209,25 +218,29 @@ Index: jellyfin-ffmpeg/libavcodec/vaapi_encode.h
...
++    // Store buffered pic.
 +    AVFifoBuffer *encode_fifo;
-+
++    // Max number of frame buffered in encoder.
```
Comment (Shadowghost) [inline debian/patches/0027-add-fixes-to-improve-the-performance-of-vaapi-encode.patch:230]:
```suggestion
+    // Max number of frames buffered by encoder.
```

**Review by Shadowghost — APPROVED** [review]
Just some small formatting stuff.

File: debian/patches/0027-add-fixes-to-improve-the-performance-of-vaapi-encode.patch (hunk 1 above)
Comment (nyanmisaka) [reply debian/patches/0027-add-fixes-to-improve-the-performance-of-vaapi-encode.patch:227]: copied from https://github.com/FFmpeg/FFmpeg/commit/e0ff86993052b49a64d434bac345e92fc149f446 i think we should leave it as is.

Comment (Shadowghost) [reply debian/patches/0027-add-fixes-to-improve-the-performance-of-vaapi-encode.patch:227]: Fine with me.

**Pull request closed without merging**